nerfstudio/data/datasets/base_dataset.py

```diff
@@ -85,10 +85,10 @@ def get_image(self, image_idx: int) -> Float[Tensor, "image_height image_width n
         """
         image = torch.from_numpy(self.get_numpy_image(image_idx).astype("float32") / 255.0)
         if self._dataparser_outputs.alpha_color is not None and image.shape[-1] == 4:
-            assert image.shape[-1] == 4
             image = image[:, :, :3] * image[:, :, -1:] + self._dataparser_outputs.alpha_color * (1.0 - image[:, :, -1:])
-        else:
-            image = image[:, :, :3]
+        elif image.shape[-1] == 3:
```

Comment:
Can you avoid doing this? It will add an overhead for cases where the opacity is not used (most cases). Instead, I would suggest keeping the original number of channels (3 or 4) if alpha_color is not set.

Reply:
done